File: GuildedRose/src/com/gildedrose/items/ConjuredItem.java
package com.gildedrose.items;

public class ConjuredItem extends ItemDecorator {
	
	// Constructors
	protected ConjuredItem(Item itemToDecorate)
	{
		super(itemToDecorate);
	}

	// Protected methods
	@Override
	protected void updateQuality()
	{
		if (this.IsSellInEnded())
		{
			this.decreaseQualityQuarce();
		}
		else
		{
			this.decreaseQualityTwice();
		}
	}

}

File: GildedRose.Net/Items/ConjuredItem.cs
namespace GildedRose.Items
{
    public class ConjuredItem : ItemDecorator
    {
    
        internal ConjuredItem(Item itemToDecorate) : base(itemToDecorate) { }
        
        // Protected methods
        protected override void UpdateQuality()
        {
            if (this.SellInEnded)
            {
                this.DecreaseQualityQuarce();
            }
            else
            {
                this.DecreaseQualityTwice();
            }
        }
        
    }
}